agentstack/generation/tool_generation.py

```diff
@@ -108,26 +115,25 @@ def remove_tool_from_tools_init(tool_data: dict, path: str = ''):
                 print(line, end='')
 
 
-def _framework_filename(framework: str, path: str = ''):
-    if framework == 'crewai':
-        return f'{path}src/crew.py'
+def add_tool_to_agent_definition(framework: str, tool_data: dict, path: str = '', agents: list[str] = []):
+    """
+        Add tools to specific agent definitions using AST transformation.
 
-    print(term_color(f'Unknown framework: {framework}', 'red'))
-    sys.exit(1)
-
-
-def add_tool_to_agent_definition(framework: str, tool_data: dict, path: str = ''):
-    filename = _framework_filename(framework, path)
-    with fileinput.input(files=filename, inplace=True) as f:
-        for line in f:
-            print(line.replace('tools=[', f'tools=[{"*" if tool_data.get("tools_bundled") else ""}tools.{", tools.".join([tool_name for tool_name in tool_data["tools"]])}, '), end='')
+        Args:
+            framework: Name of the framework
+            tool_data: Dictionary containing tool information
+                {
+                    "tools": List[str],  # List of tool names to add
+                    "tools_bundled": bool  # Whether to include tools.*
+                }
+            agents: Optional list of agent names to modify. If None, modifies all agents.
+            path: Optional path to the framework file
+        """
+    modify_agent_tools(framework, tool_data, 'add', agents, path, 'tools')
 
 
 def remove_tool_from_agent_definition(framework: str, tool_data: dict, path: str = ''):
-    filename = _framework_filename(framework, path)
-    with fileinput.input(files=filename, inplace=True) as f:
-        for line in f:
-            print(line.replace(f'{", ".join([f"tools.{tool_name}" for tool_name in tool_data["tools"]])}, ', ''), end='')
+    modify_agent_tools(framework, tool_data, 'remove', None, path, 'tools')
```

Comment:
Consider using keyword arguments here since it's not intuitive what each param is doing.

Reply:
done